ocaml/xapi/context.ml

```diff
@@ -227,8 +227,64 @@ let parent_of_origin (origin : origin) span_name =
   | _ ->
       None
 
+let attribute_helper_fn f v = Option.fold ~none:[] ~some:f v
+
+let addr_port_of_sock s =
+  match s with
+  | Unix.ADDR_UNIX "" ->
+      (None, None)
+  | Unix.ADDR_UNIX soket_name ->
+      (Some soket_name, None)
+  | Unix.ADDR_INET (addr, port) ->
+      (Some (Unix.string_of_inet_addr addr), Some (string_of_int port))
+
+let attr_of_fd s =
+  let peer_addr = s |> Unix.getpeername |> addr_port_of_sock in
```

Comment:
```
let peer_addr, peer_port =  s |> Unix.getpeername |> addr_port_of_sock
```

This saves using `fst` and `snd` because components are now named.